src/axom/core/Array.hpp

```diff
@@ -197,27 +194,32 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
    * \brief Constructor for transferring between memory spaces
    * 
    * \param [in] other The array in a different memory space to copy from
-   * \param [in] allocator_id the ID of the allocator to use (optional)
+   * \param [in] allocator_id the ID of the allocator to use (optional).
+   *
+   * \note Default value of -1 causes the new Array to use the allocator of
+   *  \a other.  If the allocator_id is specified, the new Array will be
+   *  constructed with that allocator.
+   *
+   *  If the memory space of the Array has been specified as a compile-time
+   *  template argument, and the resulting allocator does not refer to the same
+   *  memory space, the Array will instead be constructed with the default
+   *  allocator for the compile-time memory space.
    */
   template <typename OtherArrayType>
-  Array(const ArrayBase<T, DIM, OtherArrayType>& other,
-        int allocator_id = axom::detail::getAllocatorID<SPACE>());
+  Array(const ArrayBase<T, DIM, OtherArrayType>& other, int allocator_id = -1);
```

Comment:
I think we could probably split this into two constructors, one with and one without an allocator ID argument, to avoid having to define special allocator ID values.